packages/fxa-settings/src/components/FormPasswordWithBalloons/index.tsx

```diff
@@ -0,0 +1,208 @@
+                  },
+                },
+              })}
+              prefixDataTestId="new-password"
+            />
+          </FtlMsg>
+          {isNewPwdBalloonVisible && (
+            <PasswordStrengthBalloon
+              {...{
+                hasUserTakenAction: formState.dirtyFields.newPassword,
+                isTooShort: errors.newPassword?.types?.length,
+                isSameAsEmail: errors.newPassword?.types?.notEmail,
+                isCommon: errors.newPassword?.types?.uncommon,
+              }}
+            />
+          )}
+        </div>
+
+        <div className="relative">
+          <FtlMsg
+            id="form-reset-password-with-balloon-confirm-password"
+            attrs={{ label: true }}
+          >
+            <InputPassword
+              name="confirmPassword"
+              label={confirmPasswordLabel || 'Re-enter password'}
+              className="text-start"
+              onFocusCb={showConfirmPwdBalloon ? onConfirmPwdFocus : undefined}
+              onBlurCb={
+                showConfirmPwdBalloon && isConfirmPwdBalloonVisible
+                  ? onConfirmPwdBlur
+                  : undefined
+              }
+              onChange={() => {
+                if (passwordMatchErrorText) {
+                  setPasswordMatchErrorText(undefined);
+                }
+                trigger(['newPassword', 'confirmPassword']);
+              }}
+              // Password confirmation field is disabled until new password is valid
+              disabled={
+                !formState.dirtyFields.newPassword || errors.newPassword
+              }
+              hasErrors={errors.confirmPassword && passwordMatchErrorText}
+              errorText={passwordMatchErrorText}
+              inputRef={register({
+                required: true,
+                validate: (value: string) => value === getValues().newPassword,
+              })}
+              anchorStart
+              tooltipPosition="bottom"
+              prefixDataTestId="verify-password"
+            />
+          </FtlMsg>
+
+          {showConfirmPwdBalloon && isConfirmPwdBalloonVisible && (
+            <div className="input-balloon flex gap-2 items-start text-xs">
+              <IconKey className="w-10" aria-hidden="true" focusable="false" />
+              <FtlMsg id="form-password-with-balloons-confirm-pwd-balloon">
+                <p>
+                  You need this password to access any encrypted data you store
+                  with us.
+                  <br className="mb-3" />A reset means potentially losing data
+                  like passwords and bookmarks.
+                </p>
+              </FtlMsg>
+            </div>
+          )}
+        </div>
+
+        {children}
+
+        <button
+          type="submit"
+          className="cta-primary cta-xl"
+          disabled={!formState.isDirty || !formState.isValid || loading}
+        >
+          {submitButtonText || defaultButtonText}
```

Comment:
@LZoog  Can the `<FtlMsg>` be nested inside the `<button>` element, as would happens if the `FtlMsg` element is passed in as a child? I thought the FtlMsg always had to wrap another element.

Reply:
@vpomerleau It seems to be our pattern to wrap elements like that. However, I've also seen them for _inside_ other elements, so that seems fine to me.